src/workers/MetricsWorker.ts

```diff
@@ -22,6 +22,15 @@ function workerThread() {
         throw Error("Missing parentPort");
     }
 
+    const writeAndEnd = (res: ServerResponse, data: string) => {
+        if (res.writable) {
+            res.write(data);
+        }
+        if (!res.finished) {
```

Comment:
Why .end when res is not finished?
Because we have to end our response?

Reply:
Yup. `write` doesn't end it, but also calling `end` on a finished response will err and crash the process.